lib/node/nodes/kmeans.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+
+var Node = require('../node');
+var dot = require('dot');
+
+var TYPE = 'kmeans';
+
+var PARAMS = {
+    source : Node.PARAM.NODE(Node.GEOMETRY.POINT),
+    clusters : Node.PARAM.NUMBER
+};
+
+var KMeans  = Node.create(TYPE, PARAMS, {cache: false});
```

Comment:
Also this analysis is just using lat lng distances just now so we probably want to keep examples to city level otherwise its going to become less accurate